src/Mvc/Mvc.Core/src/FileStreamResult.cs

```diff
@@ -79,5 +81,42 @@ public override Task ExecuteResultAsync(ActionContext context)
             var executor = context.HttpContext.RequestServices.GetRequiredService<IActionResultExecutor<FileStreamResult>>();
             return executor.ExecuteAsync(context, this);
         }
+
+        async Task IResult.ExecuteAsync(HttpContext httpContext)
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            using (FileStream)
```

Comment:
Out of curiosity, it is possible to share this implementation with the one in FileStreamExecutor (using an internal static method)? We could make the WriteFileAsync an optional parameter to the function to help with the overload.

`static ExecuteAsync(HttpContext context, ILogger logger, Func<..., Task>)`

Reply:
Same q with the redirect result.

Reply:
That's a good idea!, I will make the change on night :)

Reply:
@pranavkm I take me more than I thought to get back into this, I have applied your suggestions, would you take a look?